src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/design_patterns/DesignPattern.java

```diff
@@ -0,0 +1,22 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.design_patterns;
+
+
+public abstract class DesignPattern {
+    public abstract String getName();
+
+    public abstract Type getType();
+
+    public abstract String getDescription();
+
+    public abstract String getExample();
+
+    public abstract RefactorProvider createRefactorProvider();
+
+    public boolean isEqualType(DesignPattern designPattern) {
+        return getType() == designPattern.getType();
+    }
+
+    public enum Type {
+        CREATIONAL, BEHAVIORAL, STRUCTURAL
```

Comment:
There are pros and cons of this approach, using an enum often breaks the OCP, of course it depends on the nature of the problem we're dealing with. If we expect that pattern types are not a subject to change very often, the code is concise, more readable and requires less unnecessary labor, to achieve the desired effect. This is what enums were invented for. However if we expect to add more design pattern families, this approach is to rigid and more dynamic solution i.e. a map should be used.